kernel/kernel-api/src/main/java/io/delta/kernel/internal/icebergcompat/IcebergCompatV3MetadataValidatorAndUpdater.java

```diff
@@ -114,7 +115,11 @@ List<IcebergCompatCheck> icebergCompatChecks() {
             CHECK_ONLY_ICEBERG_COMPAT_V3_ENABLED,
             CHECK_HAS_ALLOWED_PARTITION_TYPES,
             CHECK_HAS_NO_PARTITION_EVOLUTION,
-            CHECK_HAS_SUPPORTED_TYPE_WIDENING)
+            CHECK_HAS_SUPPORTED_TYPE_WIDENING,
+            CHECK_LITERAL_DEFAULT_VALUE)
         .collect(toList());
   }
+
+  protected static IcebergCompatCheck CHECK_LITERAL_DEFAULT_VALUE =
+      (inputContext) -> ColumnDefaults.validate(inputContext.newMetadata.getSchema(), true, true);
```

Comment:
Can you add a flag or something so that this throws an exception specific to IcebergCompatV3? This check should throw an exception that only literal values are supported for default values in iceberg compat v3 specifically